devtools/src/shared/rpc/custom-values.ts

```diff
@@ -59,29 +239,91 @@ function isNonSerializableValue(value: unknown): value is NonSerializableValue {
     value.__custom === '@@pc-non-serializable' &&
     '__type' in value &&
     typeof value.__type === 'string' &&
-    'value' in value &&
-    typeof value.value === 'string'
+    'value' in value
   )
 }
 
 function restoreClonedValue(value: NonSerializableValue) {
   if (value.__type === 'function') {
     return () => {}
   } else if (value.__type === 'symbol') {
-    return Symbol(value.value)
+    return Symbol(value.value as string)
   } else if (value.__type === 'bigint') {
     // BigInt() throws an error if the value is not a valid bigint string
     try {
-      return BigInt(value.value)
+      return BigInt(value.value as string)
+    } catch (err) {
+      return new SerializationError(`Invalid bigint value: ${value.value}`, err)
+    }
+  } else if (value.__type === 'regexp') {
+    try {
+      const regexpValue = value as NonSerializableValue_RegExp
+      const { source, flags } = regexpValue.value
+      return new RegExp(source, flags)
+    } catch (err) {
+      return new SerializationError(`Invalid regexp value: ${JSON.stringify(value.value)}`, err)
+    }
+  } else if (value.__type === 'map') {
+    const mapValue = value as NonSerializableValue_Map
+    const entries = mapValue.value.map(([k, v]) => [restoreClonedDeep(k), restoreClonedDeep(v)]) as Array<[unknown, unknown]>
+    return new Map(entries)
+  } else if (value.__type === 'set') {
+    const setValue = value as NonSerializableValue_Set
+    const values = setValue.value.map((v) => restoreClonedDeep(v))
+    return new Set(values)
+  } else if (value.__type === 'weakmap') {
+    return new WeakMap()
+  } else if (value.__type === 'weakset') {
+    return new WeakSet()
+  } else if (value.__type === 'date') {
+    try {
+      return new Date(value.value as string)
     } catch (err) {
-      console.warn(`[🍹]: Invalid bigint value: ${value.value}`, err)
-      return 0n
+      return new SerializationError(`Invalid date value: ${value.value}`, err)
     }
+  } else if (value.__type === 'arraybuffer') {
+    const bufferValue = value as NonSerializableValue_ArrayBuffer
+    return new ArrayBuffer(bufferValue.value.byteLength)
+  } else if (value.__type === 'typedarray') {
+    const typedArrayValue = value as NonSerializableValue_TypedArray
+    const buffer = new ArrayBuffer(typedArrayValue.value.byteLength)
```

Comment:
Creating a buffer of the same size is wasteful. We just need some kind of placeholder that can be understood by the devtools to display the information. Maybe use a custom class that holds the string representation?